service/src/test/java/bio/terra/pipelines/service/PipelinesServiceTest.java

```diff
@@ -170,4 +184,171 @@ void updatePipelineWorkspaceId() {
+
+    // add a required INTEGER input to the imputation pipeline
+    PipelineInputDefinition newInput =
+        new PipelineInputDefinition(pipeline.getId(), "new_integer_input", "INTEGER", true);
+
+    pipelineInputDefinitionsRepository.save(newInput);
+
+    if (shouldPassValidation) {
+      assertDoesNotThrow(() -> pipelinesService.validateInputs(pipelinesEnum, inputs));
+    } else {
+      ValidationException exception =
+          assertThrows(
+              ValidationException.class,
+              () -> pipelinesService.validateInputs(pipelinesEnum, inputs));
+      assertEquals(expectedErrorMessage, exception.getMessage());
+    }
+  }
+
+  private static Stream<Arguments> inputRequiredValidations() {
+    return Stream.of(
+        // arguments: isRequired, inputs, shouldPassValidation
+        arguments(true, new LinkedHashMap<String, Object>(Map.of("input_name", "value")), true),
+        arguments(true, new LinkedHashMap<String, Object>(), false),
+        arguments(false, new LinkedHashMap<String, Object>(Map.of("input_name", "value")), true),
+        arguments(false, new LinkedHashMap<String, Object>(), true));
+  }
+
+  @ParameterizedTest
+  @MethodSource("inputRequiredValidations")
+  void validateRequiredInputPresent(
+      Boolean isRequired, Map<String, Object> inputs, Boolean shouldPassValidation) {
+    PipelineInputDefinition inputDefinition =
+        new PipelineInputDefinition(1L, "input_name", "INTEGER", isRequired);
+    List<PipelineInputDefinition> inputDefinitions = new ArrayList<>(List.of(inputDefinition));
+
+    if (shouldPassValidation) {
+      assertTrue(pipelinesService.validateRequiredInputs(inputDefinitions, inputs).isEmpty());
+    } else {
+      assertEquals(1, pipelinesService.validateRequiredInputs(inputDefinitions, inputs).size());
+      assertEquals(
+          "input_name is required",
+          pipelinesService.validateRequiredInputs(inputDefinitions, inputs).get(0));
+    }
+  }
+
+  private static Stream<Arguments> inputTypeValidations() {
+    // error messages
+    String commonTypeErrorMessage = "input_name must be of type %s";
+    String vcfTypeErrorMessage = "input_name must be a path to a VCF file ending in .vcf.gz";
+    String arrayStringTypeErrorMessage = "input_name must be an array of non-null strings";
+    String arrayVCFTypeErrorMessage =
+        "input_name must be an array of non-null paths to VCF files ending in .vcf.gz";
+    String notNullErrorMessage = "input_name must not be null";
+    String notEmptyErrorMessage = "input_name must not be empty";
+    String emptyArrayErrorMessage = "input_name must not be an empty array";
+
+    return Stream.of(
+        // arguments: type specification, value to be tested, whether it should pass validation,
+        // expected error message if not
+        arguments("INTEGER", 123, true, null),
+        arguments("INTEGER", "123", true, null),
+        arguments("STRING", "I am a string", true, null),
```

Comment:
Worth verifying that `"\"I am a string\""` returns `"\"I am a string\""` (I'm pretty sure that with the object mapper it'll convert it to `"I am a string"`

Reply:
And actually, generally, it might make sense to have some unit testing directly on the  `PipelineInputTypesEnum` class